recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -319,16 +319,29 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-            if (self.settings.compiler == "gcc"
-                    and tools.Version(self.settings.compiler.version) < "6.0"):
-                raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
+
+    def validate(self):
+        if (self.options.shared
+            and self.settings.compiler == "gcc"
+            and tools.Version(self.settings.compiler.version) < "6.0"):
+            raise ConanInvalidConfiguration("""Doesn't support gcc5 / shared.
                 See https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-802631744""")
+        if (tools.Version(self.version) < "1.9"
+            and self.settings.os == "Macos"
+            and self.settings.arch == "armv8"):
+            raise ConanInvalidConfiguration("""This version doesn't support arm8
+                See https://github.com/aws/aws-sdk-cpp/issues/1542""")
 
     def requirements(self):
-        self.requires("aws-c-event-stream/0.1.5")
+        self.requires("aws-c-common/0.6.9")
+        if tools.Version(self.version) >= "1.9":
```

Comment:
Since this condition is repeated, could it be a property? Would be easier to read?

Reply:
Done